react/tic-tac-toe/src/utils/utils.js

```diff
@@ -11,12 +10,22 @@ const lines = [
 ];
 
 
-export const decideGameStatus = (winner, xIsNext) => winner ? `Winner: ${winner}` : `Next player: ${xIsNext ? 'X' : 'O'}`;
+export const decideGameStatus = (winner, xIsNext, playerOne, playerTwo, tie) => (
+  winner ? `Ganador: ${winner==='X' ? playerOne: playerTwo}` : tie ? `Empate` : `Le toca a: ${xIsNext ? playerOne + ' X' : playerTwo + ' O'}`);
 
 export function calculateWinner(squares) {
     for (let i = 0; i < lines.length; i++) {
       const [a, b, c] = lines[i];
-      if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) return squares[a];
+      if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]){
+        return squares[a];
+      } 
     }
     return null;
-}
\ No newline at end of file
+}
+
+export function allFill(squares) {
+  for(let i = 0; i < squares.length ; i++) {
+    if(!squares[i])return false;
+  }
+  return true;
+}
```

Comment:
I think you can do this with `every`.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every